### PR TITLE
Virtual Network API

### PR DESCRIPTION
**Pull request by isalento**
mock & live test passing. Style checks pass. At the moment resource group must exists and should be given as a parameter to live tests. -Dtest.azurecompute-arm.resourcegroup="armlivetesting"

### CODE REVIEW

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/VirtualNetworkApi.java [hunk 1]
```diff
@@ -16,60 +16,64 @@
...
-import org.jclouds.rest.annotations.BinderParam;
-import org.jclouds.rest.annotations.Fallback;
-import org.jclouds.rest.annotations.Headers;
-import org.jclouds.rest.annotations.ResponseParser;
-import org.jclouds.rest.annotations.XMLResponseParser;
-
-@Path("/services/networking")
+@Path("/subscriptions/{subscriptionid}/resourcegroups/{resourcegroup}/providers/Microsoft.Network/")
+@QueryParams(keys = "api-version", values = "2015-06-15")
 @Headers(keys = "x-ms-version", values = "{jclouds.api-version}")
```
Comment (jmspring) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/VirtualNetworkApi.java]: x-ms-version is not needed in ARM requests

Comment (isalento) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/VirtualNetworkApi.java]: Good catch. Fixed in latest.

**Comment by ritazh** [comment]
Getting the below error after running `VirtualNetworkApiLiveTest`. I did create `jcloudstest` resourcegroup ahead of time.

```
mvn -Dtest=VirtualNetworkApiLiveTest -Dtest.azurecompute-arm.identity="" -Dtest.azurecompute-arm.subscriptionid="" -Dtest.azurecompute-arm.credential="" -Dtest.azurecompute-arm.resourcegroup="jcloudstest" -Dtest.azurecompute-arm.endpoint="https://management.azure.com" -Dtest.jclouds.oauth.resource="https://management.azure.com/" -Dtest.oauth.endpoint="https://login.microsoftonline.com/[TenantId]/oauth2/token" test
```

Error:

```
<< "{"error":{"code":"ResourceGroupNotFound","message":"Resource group 'jcloudstest' could not be found."}}"
```

**Comment by ritazh** [comment]
:fist: false alarm... Looks like it's working properly. I accidentally created the resourcegroup under a different subscription. All Live tests are passing.

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/domain/VirtualNetwork.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    public abstract String id();
+
+    @Nullable
+    public abstract String etag();
+
+    public abstract String location();
+
+    public abstract VirtualNetworkProperties properties();
+
+    @SerializedNames({"name", "id", "etag", "location", "properties"})
```
Comment (ritazh) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/domain/VirtualNetwork.java]: I think you may have forgotten tags.

**Comment by ritazh** [comment]
I have added a Trello card (low priority) for `dhcpOptions` of `VirtualNetworkProperties`: https://trello.com/c/lUeVqDhJ

**Comment by ritazh** [comment]
LGTM

**Pull request merged**